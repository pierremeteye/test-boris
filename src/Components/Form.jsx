import React, { Component } from 'react';

class Form extends Component {

	constructor(props) {
		super(props)
		this.state = {
			value: '',
			newTask: ''
		}
	}

	onChange(e){
		this.setState({
			value: e.target.value,
			newTask: e.target.value
		})
	}

	onSubmit(e){
		e.preventDefault();
		this.props.getTasks(this.state.newTask);
		this.setState({
			value: ''
		});
	}


    render(){
		return <form onSubmit={this.onSubmit.bind(this)}>
			<input type="text" placeholder="Ajouter un nom" value={this.state.value} onChange={this.onChange.bind(this)}/>
			<input type="submit"/>
		</form>
	}
}

export default Form;
