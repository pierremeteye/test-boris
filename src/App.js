import React, { Component } from 'react';

// Import Custom Components
import Form from './Components/Form';
import List from './Components/List';

class App extends Component {
    constructor(props){
        super(props);
        this.state =  {
            todos: []
        }
    }
    getTasks(el){
        this.setState({todos: [...this.state.todos, el] })
    }
    onDeleteTask(el, i){
        this.state.todos.splice(i, 1);
    }
    render() {
        return (
            <div className="App">
                <h1>Todolist</h1>
                <Form getTasks={this.getTasks.bind(this)} />
                <List  
                    getTodos={this.state.todos}
                    onDeleteTask={this.onDeleteTask.bind(this)}                />
            </div>
            );
    }
}

export default App;
