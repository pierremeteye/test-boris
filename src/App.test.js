import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';

import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from "enzyme-to-json";
import sinon from "sinon";

import Form from "./Components/Form";

expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });


it("renders correctly", () => {
  const wrapper = shallow(
    <Form />
  );

  expect(wrapper).toMatchSnapshot();
});