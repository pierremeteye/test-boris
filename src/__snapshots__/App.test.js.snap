// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Ajouter un nom"
    type="text"
    value=""
  />
  <input
    type="submit"
  />
</form>
`;
